### PR TITLE
WIP Refactor Track to have a cleaner type-safe API + Nightly upgroid

### PR DESCRIPTION
**Pull request by Gankra**
This is basically #15++

### CODE REVIEW

File: containers/mkv.rs [hunk 1]
```diff
@@ -59,10 +63,10 @@ extern "C" fn read_callback(pos: c_longlong,
 
     unsafe {
         let reader: &mut Box<Box<StreamReader>> = mem::transmute(&mut user_data);
-        if reader.seek(pos, SeekStyle::SeekSet).is_err() {
+        if reader.seek(SeekFrom::Start(pos as u64)).is_err() {
             return -1
         }
-        match reader.read_at_least(len as usize, slice::from_raw_parts_mut(buf, len as usize)) {
+        match reader.read(slice::from_raw_parts_mut(buf, len as usize)) {
```
Comment (Gankra) [inline containers/mkv.rs]: Wait, I don't think this is right. Need to loop and keep asking for bytes until it's full. Seems to be working in my testing but probably super fragile.

Do you think we should have a little utils module to have this function?

**Comment by Gankra** [comment]
@pcwalton I've added a write_to_full util and added a dependency on `byteorder` to get back endianess methods. rust-media itself doesn't use any old_\* now. Only the example needs old_io to `sleep`, which currently has no replacement.

File: utils.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+use std::io::{Error, ErrorKind, Result};
```
Comment (pcwalton) [inline utils.rs:1]: Please add a license header to this. (You can copy it from any existing `.rs` file.)

Comment (Gankra) [inline utils.rs:1]: done

**Comment by pcwalton** [comment]
Looks good, squash and I'll merge. Thanks!

**Comment by Gankra** [comment]
squashed

**Pull request merged**